src/main/java/com/example/accommodationbookingapp/controller/BookingController.java

```diff
@@ -0,0 +1,110 @@
+@RequiredArgsConstructor
+public class BookingController {
+    private final BookingService bookingService;
+
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PostMapping
+    @Operation(summary = "Create a new booking")
+    @ResponseStatus(HttpStatus.CREATED)
+    public BookingResponseDto createBooking(
+            @RequestBody @Valid CreateBookingRequestDto requestDto,
+            Authentication authentication) {
+        User user = (User) authentication.getPrincipal();
+        return bookingService.createBooking(user.getId(), requestDto);
+    }
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @GetMapping()
+    @Operation(
+            summary = "Retrieves bookings",
+            description = "Retrieves bookings based on user ID and their status"
+    )
+    public List<BookingResponseDto> getBookingsByUserIdAndStatus(
+            @RequestParam(name = "user_id") Optional<Long> optionalId,
+            @RequestParam(name = "status") Booking.Status status,
+            Pageable pageable
+    ) {
+        return optionalId.map(id ->
+                        bookingService.getBookingsByUserIdAndStatus(id, status, pageable))
+                .orElseGet(() -> bookingService.getBookingsByStatus(status, pageable));
+    }
+
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/my")
+    @Operation(summary = "Retrieves user bookings")
+    public List<BookingResponseDto> getUserBookings(
+            Authentication authentication,
+            Pageable pageable
+    ) {
+        User user = (User) authentication.getPrincipal();
+        return bookingService.getBookingHistory(user.getId(), pageable);
+    }
+
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @Operation(
+            summary = "Get the booking",
+            description = "Provides information about a specific booking"
+    )
+    @GetMapping("/{id}")
+    public BookingResponseDto getBooking(
+            Authentication authentication,
+            @PathVariable(name = "id") Long bookingId) {
+        User user = (User) authentication.getPrincipal();
+        return bookingService.getBookingByIdAndUserId(user.getId(), bookingId);
+    }
+
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PatchMapping("/{id}")
+    @Operation(
+            summary = "Update the booking",
+            description = "Allows users to update their booking details")
+    public BookingResponseDto updateBooking(
+            Authentication authentication, @PathVariable(name = "id") Long bookingId,
+            @RequestBody @Valid UpdateBookingRequestDto requestDto) {
+        User user = (User) authentication.getPrincipal();
+        return bookingService.updateBookingDetails(user.getId(), bookingId, requestDto);
+    }
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @Operation(
+            summary = "Delete the booking",
+            description = "Enables the cancellation of bookings"
+    )
```

Comment:
More consistent variant:
```suggestion
    @PreAuthorize("hasRole('USER')")
    @DeleteMapping("/{id}")
    @Operation(
            summary = "Delete the booking",
            description = "Enables the cancellation of bookings"
    )
    @ResponseStatus(HttpStatus.NO_CONTENT)
```